expanded observability note with config settings getting replaced (#6518)
open-cluster-management/rhacm-docs: observability/observability_enable.adoc

- When observability is enabled or disabled on {ocp-short} managed clusters that are managed by {product-title-short}, the observability endpoint operator updates the `cluster-monitoring-config` config map by adding additional `alertmanager` configuration that automatically restarts the local Prometheus.
- The observability endpoint operator updates the `cluster-monitoring-config` config map by adding additional `alertmanager` configurations that automatically restart the local Prometheus. Therefore, when you insert the `alertmanager` configuration in the {ocp-short} managed cluster, the configuration removes the settings that relate to the retention of the Prometheus metrics. 

> - When you import an {ocp-short} managed cluster onto the hub cluster, {product-title-short} overwrites the `retentionSize` configuration under the `prometheusK8s` section of the config map. 

Reviewer: Let's not use "under" --- reason:

Let's say we were to run an accessibility test on content, this is a word that could get flagged as a directional word that could confuse a user with a screen reader...then one would have to go justify the use of the word. It's kinder to the user not to use it and it reduces overhead to change or explain.

Reviewer: I sometimes change to "from the ___section" "Within the section" something that is not directional.

> - When you import an {ocp-short} managed cluster onto the hub cluster, {product-title-short} removes all the configuration settings from the `cluster-monitoring-config` config map of the {ocp-short} managed cluster. When the configuration settings are removed, {product-title-short} replaces it with the settings from the hub cluster.  

Reviewer 2: This is not correct. The only thing being overwritten is the `retentionSize` configuration under the `prometheusK8s` section of the config map. 

We should probably document this as a known issue.

Reviewer 3: @douglascamata I have a case attached to the following knowledge base article [1]
that details an issue a CU had with additional alertmanager secrets [2]being removed when importing a managed cluster into ACM - hence the ask to expand this statement in the documentation 

[1] - https://access.redhat.com/solutions/7070731 
[2] - https://access.redhat.com/documentation/de-de/openshift_container_platform/4.14/html-single/monitoring/index#monitoring-adding-a-secret-to-the-alertmanager-configuration_configuring-the-monitoring-stack

Reviewer 2: @coffeegoesincodecomesout let me double check what's the situation for that alertmanager configuration. The root cause is more generic and can involve other fields of the config map being reset to their default values.

Reviewer 2: @coffeegoesincodecomesout in the `alertmanagerMain` section of the config map, these are the fields that **will not** be overwritten:

* `enabled`
* `enableUserAlertmanagerConfig`
* `logLevel`
* `nodeSelector`
* `tolerations`
* `resources`
* `volumeClaimTemplate`

Everything else will be gone. As you can see, the `secrets` key is not one of the configurations that will stay.

Reviewer 2: @coffeegoesincodecomesout it's hard to document everything that will or will not be overwritten from this config map, as it's quite extensive. 

Basically the MCO is locked to the configuration from the commit `20fcb9f6ef4e00ccde05a13347bddfabe24e8c4c` of the CMO: https://github.com/openshift/cluster-monitoring-operator/blob/20fcb9f6ef4e00ccde05a13347bddfabe24e8c4c/pkg/manifests/config.go#L95. Only fields present in that version of the `ClusterMonitoringConfiguration` will persist.

> - When you import an {ocp-short} managed cluster into the hub cluster, {product-title-short} removes all the configuration settings in that {ocp-short} managed cluster `cluster-monitoring-config` configmap. After removing the configuration settings, {product-title-short} replaces it with the settings passed down from the hub cluster.  

Reviewer 4: Can we use something different from "passed down". Does the console take the action of replacing the configuration settings or does the user have to remove it? Removing anthropomorphism and removed directional phrase

```suggestion
- When you import an {ocp-short} managed cluster onto the hub cluster, {product-title-short} removes all the configuration settings from the `cluster-monitoring-config` config map of the {ocp-short} managed cluster. When the configuration settings are removed, {product-title-short} replaces it with the settings from the hub cluster.  
```
